api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -347,6 +347,17 @@ default Table registerTable(TableIdentifier identifier, String metadataFileLocat
     throw new UnsupportedOperationException("Registering tables is not supported");
   }
 
+  /**
+   * Reset the table with a meta file.
+   *
+   * @param identifier a table identifier
+   * @param metadataFileLocation the location of a metadata file
+   * @return a Table instance
+   */
+  default Table resetTable(TableIdentifier identifier, String metadataFileLocation) {
```

Comment:
Do we really need to add another method to `Catalog`? What is the difference between this and `dropTable` followed by `registerTable`?

Reply:
But It may have a time gap, when there are consecutive queries. And it doesn't seem guaranteed to be a transactional operation